inception-scheduling/src/main/java/de/tudarmstadt/ukp/inception/scheduling/SchedulingService.java

```diff
@@ -85,13 +85,15 @@ private void afterExecute(Runnable aRunnable, Throwable aThrowable)
         return result;
     }
 
-    public void enqueue(Task aTask)
+    public synchronized void enqueue(Task aTask, String aOrigin)
     {
         if (getScheduledTasks().contains(aTask)) {
             log.debug("Task already in queue: {}", aTask);
             return;
         }
 
+        log.debug("Enqueuing task [{}] - Source: [{}]", aTask, aOrigin);
```

Comment:
I think it would be good if the task would remember why it was triggered so we can even log this info when the task completes (asynchronously).

Reply:
Done